lib/smart-app.js

```diff
@@ -429,24 +492,146 @@ module.exports = class SmartApp {
 	 */
 
 	/**
-	 * @callback EventCallback
+	 * @callback ModeEventCallback
+	 * @param context { import('./util/endpoint-context') }
+	 * @param {ModeEvent} modeEvent
+	 */
+
+	/**
+	 * @callback DeviceEventCallback
+	 * @param context { import('./util/endpoint-context') }
+	 * @param {DeviceEvent} deviceEvent
+	 */
+
+	/**
+	 * @callback TimerEventCallback
 	 * @param context { import('./util/endpoint-context') }
-	 * @param {ModeEvent|DeviceEvent|TimerEvent} event
-	 * @returns {EventCallback}
+	 * @param {TimerEvent} timerEvent
 	 */
 
 	/**
-	 * Handler for named subscriptions to events
+	 * @callback HubHealthEventCallback
+	 * @param context { import('./util/endpoint-context') }
+	 * @param {HubHealthEvent} hubHealthEvent
+	 */
+
+	/**
+	 * @callback DeviceHealthEventCallback
+	 * @param context { import('./util/endpoint-context') }
+	 * @param {DeviceHealthEvent} deviceHealthEvent
+	 */
+
+	/**
+	 * @callback DeviceLifecycleEventCallback
+	 * @param context { import('./util/endpoint-context') }
+	 * @param {DeviceLifecycleEvent} deviceLifecycleEvent
+	 */
+
+	/**
+	 * @callback SecurityArmStateEventCallback
+	 * @param context { import('./util/endpoint-context') }
+	 * @param {SecurityArmStateEvent} securityArmStateEvent
+	 */
+
+	/**
+	 * Internal handler for named subscriptions to events
 	 *
 	 * @param {String} name Provide the name matching a created subscription
-	 * @param {EventCallback} callback Callback handler object
+	 * @param {*} callback Callback handler object
+	 * @param {String} [eventType=undefined] The type of event
 	 * @returns {SmartApp} SmartApp instance
 	 */
-	subscribedEventHandler(name, callback) {
+	_subscribedEventHandler(name, callback, eventType = undefined) {
```

Comment:
@erodewald  this is a breaking change isn't it?  Apps using {{subscribedEventHandler()}} now will stop working. Seems like we should leave it as is. at least for now.

The original idea was not to tie handlers to a single type of event, but in practice that's the most frequent use-case, so, I guess its OK. IMO it would be cleaner to have just one method for subscribed handlers, but the failure to pass the name back makes that unsupportable for now.

Reply:
Echoing what I said to you over chat – un-renaming this gives backward compatibility.